### ISSUE TITLE
Runtime Error!

### ISSUE THREAD

**Issue by JamesCutter** [elishacloud/Silent-Hill-2-Enhancements#59]
I followed every single installation step (the updated steps) and Im getting Microsoft visual C runtime errors. I tried installing the 2010 version but it says I already have a newer version installed. I dont know what to do anymore. Running the game as admin doesnt fix it either.

Runtime Error!

Program C:\Program Files (x86)\Konami\Silent Hill 2\sh2pc.exe

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application’s support team for more information﻿

Help pls

**Comment by Polymega** [comment]
Hi @JamesCutter , 

If you set `d3d8to9 = 0` in the `d3d8.ini` file does the game then boot up without issue?

**Comment by pwnisher317** [comment]
im having the same problem and no that doesnt help unless you have directx8 and even then i dont think it works.

**Comment by Polymega** [comment]
The game natively runs on DirectX 8. 

If you set `AutoUpdateModule = 0` does the game run then?

**Comment by Polymega** [comment]
Please share your `d3d8.log` file here.

**Comment by pwnisher317** [comment]
[d3d8.log](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2527455/d3d8.log)

**Comment by Polymega** [comment]
How long did you wait when the screen was solid black before trying to force close? 
Are you also using the WineD3D files?

**Comment by Polymega** [comment]
Okay, keep `AutoUpdateModule` disabled and allow up to two minutes for the game to start upon launching it. The fact you got a black screen and **not** the C++ error when disabling the `AutoUpdateModule` is a good thing. 

It can take a while to boot up, especially when using ReShade on non-gaming specific laptops. There's no one specific reason why some people can boot up the game quicker than others. Hardware, of course plays a big factor, but there can be other unknown reasons as well. Try letting it sit for a minute or two next time.

**Comment by pwnisher317** [comment]
no, i got the same error as the guy that posted this, that the application requested the runtime to terminate or something like that

**Comment by Polymega** [comment]
According to your log file it seems to be getting hung up on the modupdater, as that's the next thing that would load after the Widescreen Fix & sh2proxy module. 

Have you tried keeping `AutoUpdateModule` disabled in the `d3d8.ini` file, launching the game, and waiting up to two minutes once you receive the black screen as you said you received prior?

**Comment by pwnisher317** [comment]
yeah it didnt work still

**Comment by Polymega** [comment]
Also, referencing this post here: https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/46#issuecomment-424296158

You can also try downloading a modified `d3d8.ini` file Elisha made which is [found here](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/2409446/d3d8.zip). If that doesn't work as-is, trying disabling the modupdater in that `d3d8.ini` file. If that doesn't work, and as Elisha suggests in that thread, try disabling a fix one-at-a-time while relaunching the game each time to see if a particular fix is causing your issues.

**Comment by pwnisher317** [comment]
eh, thats too much effort. id rather play it as is than have to go through all that trouble. took me long enough to download and find where all this stuff goes

**Comment by pwnisher317** [comment]
i fixed it, and this seems a bit odd but it worked. i decided to put the stuff thats supposed to be in the sh2e folder in the data folder and everything seemed to work fine. for example the contents of the op folder went into the op folder in the datas bg folder and such. or whatever folders go where.

**Comment by pwnisher317** [comment]
okay now ive got the game to work but the fully rendered cg cutscenes are extremely laggy and the music is choppy, and then the regular in game cutscenes are slow andd not in sync.... im having way more trouble with this than i should be

**Comment by Polymega** [comment]
Try disabling fixes one at a time, while relaunching the game each time, to see if that boosts performance. 

I also can't promise you won't run into any issues by moving all the modded files from the `sh2e` folder to the main `data` folder. It would have been ideal to do the recommended fix first before trying that. 

The fog fix, setting the noise grain filter too high, and using ReShade and/or WineD3D are the three biggest culprits for performance hits. I have to disable the fog fix on my laptop otherwise the game runs too slow during outside segments.

**Comment by JamesCutter** [comment]
disabling AutoUpdateModule fixed the error for me. But for some reason the games resolution is not 1920x 1080 even thought I edited the resolution in the d3d8.ini. Do u guys know how to make the game run in 1080p ?

**Comment by Polymega** [comment]
Hi @JamesCutter ,
I'm glad to hear that. 

> But for some reason the games resolution is not 1920x 1080 even thought I edited the resolution in the d3d8.ini.

It should be running at 1080p then. Because I believe you mean that when you check the game's resolution from the in-game Options menu it shows something different, correct? (Most likely 640 x 480?) 

If that's the case simply ignore whatever the resolution says from the game's Options menu. That resolution is overrode from the `d3d8.ini` file you adjusted. 

Elisha made mention that it could be possible to display the resolution that you set in the .ini file on the game's option menu (but you'd be unable to change the resolution from within the game still). But this would be a low-priority thing to do and hasn't been looked in to yet.

**Comment by JamesCutter** [comment]
Oh ok that makes sense. Ingame the game definetly looks 1080p, I just got confused by the Options. Thanks for your reply Im super happy now :)
